src/idl_gen_dlang.cpp

```diff
@@ -0,0 +1,706 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// independent from idl_parser, since this code is not needed for most clients
+
+#include <list>
+#include <algorithm>
+
+#include "flatbuffers/flatbuffers.h"
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+#include "flatbuffers/code_generators.h"
+
+namespace flatbuffers {
+
+namespace dlang {
+
+class DlangGenerator : public BaseGenerator {
+public:
+ DlangGenerator(const Parser &parser, const std::string &path,
+              const std::string &file_name)
+     : BaseGenerator(parser, path, file_name, "", "."),
+       cur_name_space_(nullptr){}
+
+   bool generate(){
+       std::string one_file_code;
+       cur_name_space_ = parser_.namespaces_.back();
+       for (auto it = parser_.enums_.vec.begin(); it != parser_.enums_.vec.end();
```

Comment:
not sure which C++ version does it use, but maybe `for (auto& it : parser_.enums_.vec) {` ?